src/core/layout/qgslayoutatlas.cpp

```diff
@@ -395,6 +395,21 @@ bool QgsLayoutAtlas::endRender()
   return true;
 }
 
+void QgsLayoutAtlas::filterCoverageLayer()
+{
+  if ( mFilterFeatures && !mFilterExpression.isEmpty() )
+  {
+    mOriginalFilterExpression = mCoverageLayer->subsetString();
+    mCoverageLayer->setSubsetString( mFilterExpression );
```

Comment:
There's a subtle difference between an expression based filter and a layer subset string. Subset strings do NOT use qgis expressions, and instead use the language of whatever filtering is available on the actual data source backend. So we can't use them interchangeably like you do here

Reply:
Right.

Reply:
I came up with these methods in order to have a way to filter the coverage layer before the exporting of atlas via the processing algorithms.